modules/sops/default.nix

```
@@ -144,17 +141,17 @@ in {
   };
   config = mkIf (cfg.secrets != {}) {
     assertions = [{
-      assertion = cfg.gnupgHome != null -> cfg.sshKeyPaths == [];
-      message = "Configuration options sops.gnupgHome and sops.sshKeyPaths cannot be set both at the same time";
+      assertion = (cfg.gnupgHome == null) != (cfg.sshKeyPaths == []);
+      message = "Exactly one of sops.gnupgHome and sops.sshKeyPaths must be set";
+    }] ++ concatLists (mapAttrsToList (name: secret: [{
+      assertion = builtins.pathExists secret.sopsFile;
+      message = "Cannot find path '${secret.sopsFile}' set in sops.secrets.${strings.escapeNixIdentifier name}.sopsFile";
     } {
-      assertion = cfg.gnupgHome == null -> cfg.sshKeyPaths != [];
-      message = "Either sops.sshKeyPaths and sops.gnupgHome must be set";
-    }] ++ map (name: let
-      inherit (cfg.secrets.${name}) sopsFile;
-    in {
-      assertion = cfg.validateSopsFiles -> builtins.isPath sopsFile;
-      message = "${sopsFile} is not in the nix store. Either add it to the nix store or set `sops.validateSopsFiles` to false";
-    }) (builtins.attrNames cfg.secrets);
+      assertion = cfg.validateSopsFiles ->
+        builtins.isPath secret.sopsFile ||
+        (builtins.isString secret.sopsFile && hasPrefix builtins.storeDir secret.sopsFile);
```

I find this hard to understand. I assume this is how it will be evaluated?

```suggestion
      assertion = (cfg.validateSopsFiles -> builtins.isPath secret.sopsFile) ||
        (builtins.isString secret.sopsFile && hasPrefix builtins.storeDir secret.sopsFile);
```

---

`a -> b` is the same as `!a || b`, and `||` is associative, which means it doesn't matter where you put the parentheses. If you want to have them anyway for clarity, I'd put them on the right instead:
```nix
assertion = cfg.validateSopsFiles -> (builtins.isPath secret.sopsFile ||
        (builtins.isString secret.sopsFile && hasPrefix builtins.storeDir secret.sopsFile));
```

---

Yes please.